src/SFML/Network/TcpSocket.cpp

```diff
@@ -242,11 +242,11 @@ Socket::Status TcpSocket::send(const void* data, std::size_t size, std::size_t&
     }
 
     // Loop until every byte has been sent
-    int result = 0;
-    for (sent = 0; sent < size; sent += result)
+    ssize_t result = 0;
+    for (sent = 0; sent < size; sent += static_cast<std::size_t>(result))
     {
         // Send a chunk of data
-        result = ::send(getHandle(), static_cast<const char*>(data) + sent, static_cast<int>(size - sent), flags);
+        result = ::send(getHandle(), static_cast<const char*>(data) + sent, size - sent, flags);
```

Comment:
Here any with all the other socket function we have a fun issue, that the signatures are different on Unix and on Windows.

The `size` parameter is taken as `int` [on Windows](https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-send) and as `size_t` [on Linux](https://linux.die.net/man/2/send).

How do we want to solve this?

- Silence the warning somehow (anyone know how?)
- Use `#ifdef` for Unix and Windows
- Move the functions into the dedicated pImpl classes
- ???

Reply:
There is already `priv::SocketImpl::AddrLength`, can we not use it?
```cpp
static_cast<priv::SocketImpl::AddrLength>(size - sent)
```

In case this type is not correct, let's add another typedef rather than outsourcing the function.

Reply:
`AddrLength` is differently defined for Unix, but I added a new typedef `Size`.
Let me know what you think!

Reply:
Looks good, thanks! 👍🏼